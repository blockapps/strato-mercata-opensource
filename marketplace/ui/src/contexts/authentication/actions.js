import RestStatus from 'http-status-codes';
import { apiUrl, HTTP_METHODS } from '../../helpers/constants';

const actionDescriptors = {
  check: 'auth/check',
  checkSuccessful: 'auth/check_successful',
  checkFailed: 'auth/check_failed',
  checkFailedInternalServerError: 'auth/check_failed_internal_server_error',
  fetchUsers: 'FETCH_USERS',
  fetchUsersSuccessful: 'FETCH_USERS_SUCCESSFUL',
  fetchUsersFailed: 'FETCH_USERS_FAILED',
  logoutCheck: 'auth/logout_check',
  logoutSuccessful: 'auth/logout_success',
  logoutFailed: 'auth/logout_failed',
};

const actions = {
  check: async (dispatch) => {
    dispatch({ type: actionDescriptors.check });
    try {
      let response = await fetch(`${apiUrl}/users/me`, {
        method: HTTP_METHODS.GET,
        credentials: 'same-origin',
      });
      const body = await response.json();
      if (response.status === RestStatus.INTERNAL_SERVER_ERROR) {
        {
          dispatch({
            type: actionDescriptors.checkFailedInternalServerError,
            error: body.error,
          });
          return;
        }
      } else if (
        response.status === RestStatus.UNAUTHORIZED ||
        response.status === RestStatus.FORBIDDEN
      ) {
        dispatch({
          type: actionDescriptors.checkFailed,
          payload: body.error.loginUrl,
        });
        return;
      } else if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.checkSuccessful,
          payload: { ...body.data },
        });
        return;
      }
      dispatch({ type: actionDescriptors.checkFailed, payload: undefined });
    } catch (err) {
      dispatch({ type: actionDescriptors.checkFailed, payload: undefined });
    }
  },
  fetchUsers: async (dispatch) => {
    dispatch({ type: actionDescriptors.fetchUsers });
    try {
      const response = await fetch(`${apiUrl}/users`, {
        method: HTTP_METHODS.GET,
        credentials: 'same-origin',
      });
      const body = await response.json();
      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.fetchUsersSuccessful,
          payload: body.data,
        });
        return;
      } else if (response.status === RestStatus.UNAUTHORIZED) {
        dispatch({
          type: actionDescriptors.fetchUsersFailed,
          error: 'Unauthorized while fetching users',
        });
        window.location.href = body.error.loginUrl;
      }
      dispatch({
        type: actionDescriptors.fetchUsersFailed,
        payload: undefined,
      });
    } catch (err) {
      dispatch({
        type: actionDescriptors.fetchUsersFailed,
        payload: undefined,
      });
    }
  },
  logout: async (dispatch) => {
    dispatch({ type: actionDescriptors.logoutCheck });
    try {
      const response = await fetch(`${apiUrl}/authentication/logout`, {
        method: HTTP_METHODS.GET,
        credentials: 'same-origin',
      });
      const body = await response.json();
      if (response.status === RestStatus.OK) {
        window.location.href = body.data.logoutUrl;
        localStorage.removeItem('loginCount');
        dispatch({
          type: actionDescriptors.logoutSuccessful,
          payload: body.data.logoutUrl,
        });
        return;
      } else if (response.status === RestStatus.UNAUTHORIZED) {
        dispatch({
          type: actionDescriptors.logoutFailed,
          error: 'Unauthorized while logging out',
        });
        window.location.href = body.error.loginUrl;
      }
      dispatch({ type: actionDescriptors.logoutFailed, payload: undefined });
    } catch (err) {
      dispatch({ type: actionDescriptors.logoutFailed, payload: undefined });
    }
  },
};

export { actionDescriptors, actions };
