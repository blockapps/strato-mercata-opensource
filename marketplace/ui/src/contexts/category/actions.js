import RestStatus from 'http-status-codes';
import { apiUrl, HTTP_METHODS } from '../../helpers/constants';

const actionDescriptors = {
  createCategory: 'create_category',
  createCategorySuccessful: 'create_category_successful',
  createCategoryFailed: 'create_category_failed',
  fetchCategory: 'fetch_categorys',
  fetchCategorySuccessful: 'fetch_category_successful',
  fetchCategoryFailed: 'fetch_category_failed',
  resetMessage: 'reset_message',
  setMessage: 'set_message',
};

const actions = {
  resetMessage: (dispatch) => {
    dispatch({ type: actionDescriptors.resetMessage });
  },

  setMessage: (dispatch, message, success = false) => {
    dispatch({ type: actionDescriptors.setMessage, message, success });
  },

  createCategory: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.createCategory });

    try {
      const response = await fetch(`${apiUrl}/category`, {
        method: HTTP_METHODS.POST,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.createCategorySuccessful,
          payload: body.data,
        });
        actions.setMessage(dispatch, 'Category created successfully', true);
        return true;
      }

      dispatch({
        type: actionDescriptors.createCategoryFailed,
        error: 'Error while creating Category',
      });
      actions.setMessage(dispatch, 'Error while creating Category');
      return false;
    } catch (err) {
      dispatch({
        type: actionDescriptors.createCategoryFailed,
        error: 'Error while creating Category',
      });
      actions.setMessage(dispatch, 'Error while creating Category');
    }
  },

  fetchCategories: async (dispatch) => {
    dispatch({ type: actionDescriptors.fetchCategory });

    try {
      const response = await fetch(`${apiUrl}/category`, {
        method: HTTP_METHODS.GET,
      });

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.fetchCategorySuccessful,
          payload: body.data,
        });
        return;
      } else if (response.status === RestStatus.INTERNAL_SERVER_ERROR) {
        dispatch({
          type: actionDescriptors.fetchCategoryFailed,
          error: 'Error while fetching category',
        });
        return false;
      } else if (response.status === RestStatus.UNAUTHORIZED) {
        dispatch({
          type: actionDescriptors.fetchCategoryFailed,
          error: 'Unauthorized while fetching categories',
        });
        window.location.href = body.error.loginUrl;
      }

      dispatch({
        type: actionDescriptors.fetchCategoryFailed,
        error: body.error,
      });
    } catch (err) {
      dispatch({
        type: actionDescriptors.fetchCategoryFailed,
        error: 'Error while fetching category',
      });
    }
  },
};

export { actionDescriptors, actions };
