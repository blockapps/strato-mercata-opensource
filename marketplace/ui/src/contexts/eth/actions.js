import RestStatus from 'http-status-codes';
import { apiUrl, HTTP_METHODS } from '../../helpers/constants';

const actionDescriptors = {
  resetMessage: 'reset_message',
  setMessage: 'set_message',
  fetchETHSTAddress: 'fetch_ethst_address',
  fetchETHSTAddressSuccessful: 'fetch_ethst_address_successful',
  fetchETHSTAddressFailed: 'fetch_ethst_address_failed',
  addHash: 'add_hash',
  addHashSuccessful: 'add_hash_successful',
  addHashFailed: 'add_hash_failed',
};

const actions = {
  resetMessage: (dispatch) => {
    dispatch({ type: actionDescriptors.resetMessage });
  },

  setMessage: (dispatch, message, success = false) => {
    dispatch({ type: actionDescriptors.setMessage, message, success });
  },

  fetchETHSTAddress: async (dispatch) => {
    dispatch({ type: actionDescriptors.fetchETHSTAddress });
    try {
      let response = await fetch(`${apiUrl}/tokens/address`, {
        method: HTTP_METHODS.GET,
        credentials: 'same-origin',
      });

      const body = await response.json();
      if (
        response.status === RestStatus.UNAUTHORIZED ||
        response.status === RestStatus.FORBIDDEN
      ) {
        dispatch({
          type: actionDescriptors.fetchETHSTAddressFailed,
          payload: 'Error while fetching ETHST address',
        });
        return null;
      }

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.fetchETHSTAddressSuccessful,
          payload: body?.data,
        });
        return body.data;
      }

      dispatch({
        type: actionDescriptors.fetchETHSTAddressFailed,
        payload: 'Error while fetching ETHST address',
      });
      return null;
    } catch (err) {
      dispatch({
        type: actionDescriptors.fetchETHSTAddressFailed,
        payload: 'Error while fetching ETHST address',
      });
      return null;
    }
  },

  addHash: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.addHash });

    try {
      const response = await fetch(`${apiUrl}/tokens/addHash`, {
        method: HTTP_METHODS.POST,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.addHashSuccessful,
          payload: body.data,
        });
        actions.setMessage(dispatch, 'Hash added successfully', true);
        return true;
      } else if (response.status === RestStatus.CONFLICT) {
        dispatch({
          type: actionDescriptors.addHashFailed,
          error: body.error.message,
        });
        actions.setMessage(dispatch, body.error.message);
        return false;
      } else if (response.status === RestStatus.INTERNAL_SERVER_ERROR) {
        dispatch({
          type: actionDescriptors.addHashFailed,
          error: 'Error while adding Hash',
        });
        actions.setMessage(dispatch, 'Error while adding Hash');
        return false;
      } else if (response.status === RestStatus.UNAUTHORIZED) {
        dispatch({
          type: actionDescriptors.addHashFailed,
          error: 'Unauthorized while adding Hash',
        });
        window.location.href = body.error.loginUrl;
      }

      dispatch({
        type: actionDescriptors.addHashFailed,
        error: body.error,
      });
      actions.setMessage(dispatch, body.error);
      return false;
    } catch (err) {
      dispatch({
        type: actionDescriptors.addHashFailed,
        error: 'Error while adding Hash',
      });
      actions.setMessage(dispatch, 'Error while adding Hash');
    }
  },
};

export { actionDescriptors, actions };
