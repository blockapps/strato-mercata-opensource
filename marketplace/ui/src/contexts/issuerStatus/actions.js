import RestStatus from 'http-status-codes';
import { apiUrl, HTTP_METHODS } from '../../helpers/constants';

const actionDescriptors = {
  setMessage: 'set_message',
  resetMessage: 'reset_message',
  requestReview: 'request_review',
  requestReviewSuccessful: 'request_review_success',
  requestReviewFailed: 'request_review_failed',
  authorizeIssuer: 'authorize_issuer',
  authorizeIssuerSuccessful: 'authorize_issuer_success',
  authorizeIssuerFailed: 'athorize_issuer_failed',
  deauthorizeIssuer: 'deauthorize_issuer',
  deauthorizeIssuerSuccessful: 'deauthorize_issuer_success',
  deauthorizeIssuerFailed: 'deathorize_issuer_failed',
  modifyAdmin: 'modify_admin',
  modifyAdminSuccessful: 'modify_admin_successful',
  modifyAdminFailed: 'modify_admin_failed',
};

const actions = {
  resetMessage: (dispatch) => {
    dispatch({ type: actionDescriptors.resetMessage });
  },
  setMessage: (dispatch, message, success = false) => {
    dispatch({ type: actionDescriptors.setMessage, message, success });
  },
  requestReview: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.requestReview });
    try {
      const response = await fetch(`${apiUrl}/issuerstatus/requestReview`, {
        method: HTTP_METHODS.POST,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });
      const body = await response.json();
      if (response.status == RestStatus.OK) {
        dispatch({ type: actionDescriptors.requestReviewSuccessful });
        actions.setMessage(dispatch, 'Successfully requested review', true);
        return body;
      } else {
        dispatch({ type: actionDescriptors.requestReviewFailed });
        actions.setMessage(dispatch, body.error, false);
      }
    } catch (e) {
      dispatch({ type: actionDescriptors.requestReviewFailed });
      actions.setMessage(
        dispatch,
        'Error occurred while requesting review: ' + e.message,
        false
      );
    }
  },
  authorizeIssuer: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.authorizeIssuer });
    try {
      const response = await fetch(`${apiUrl}/issuerstatus/authorizeIssuer`, {
        method: HTTP_METHODS.POST,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });
      const body = await response.json();
      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.authorizeIssuerSuccessful,
          payload: body.data,
        });
        actions.setMessage(dispatch, 'Successfully authorized issuer', true);
        return body;
      } else {
        dispatch({ type: actionDescriptors.authorizeIssuerFailed });
        actions.setMessage(dispatch, body.error, false);
      }
    } catch (err) {
      dispatch({ type: actionDescriptors.authorizeIssuerFailed });
      actions.setMessage(dispatch, err.message, false);
    }
  },
  deauthorizeIssuer: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.deauthorizeIssuer });
    try {
      const response = await fetch(`${apiUrl}/issuerstatus/deauthorizeIssuer`, {
        method: HTTP_METHODS.POST,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });
      const body = await response.json();
      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.deauthorizeIssuerSuccessful,
          payload: body.data,
        });
        actions.setMessage(dispatch, 'Successfully deauthorized issuer', true);
        return body;
      } else {
        dispatch({ type: actionDescriptors.deauthorizeIssuerFailed });
        actions.setMessage(dispatch, body.error, false);
      }
    } catch (err) {
      dispatch({ type: actionDescriptors.deauthorizeIssuerFailed });
      actions.setMessage(dispatch, err.message, false);
    }
  },
  modifyAdmin: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.modifyAdmin });
    try {
      const response = await fetch(`${apiUrl}/issuerstatus/admin`, {
        method: HTTP_METHODS.POST,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });
      const body = await response.json();
      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.modifyAdminSuccessful,
          payload: body.data,
        });
        actions.setMessage(
          dispatch,
          "Successfully updated user's admin status",
          true
        );
        return body;
      } else {
        dispatch({ type: actionDescriptors.modifyAdminFailed });
        actions.setMessage(dispatch, body.error, false);
      }
    } catch (err) {
      dispatch({ type: actionDescriptors.modifyAdminFailed });
      actions.setMessage(dispatch, err.message, false);
    }
  },
};

export { actionDescriptors, actions };
