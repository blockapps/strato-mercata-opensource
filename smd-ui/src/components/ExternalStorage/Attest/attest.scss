.attest-result ul {
  list-style-type: none;
}

.attest-result li {
  color: lightgreen; 
}

.attest-result li:before {
  font-family: 'FontAwesome';
  content: '\F00C';
  margin: 0 5px 0 0px;
  color: lightgreen;
  font-size: 20px;
}