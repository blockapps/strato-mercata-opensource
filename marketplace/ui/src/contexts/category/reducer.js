import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createCategory:
      return {
        ...state,
        isCreateCategorySubmitting: true,
      };
    case actionDescriptors.createCategorySuccessful:
      return {
        ...state,
        category: action.payload,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.createCategoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.fetchCategory:
      return {
        ...state,
        iscategorysLoading: true,
      };
    case actionDescriptors.fetchCategorySuccessful:
      return {
        ...state,
        categorys: action.payload,
        iscategorysLoading: false,
      };
    case actionDescriptors.fetchCategoryFailed:
      return {
        ...state,
        error: action.error,
        iscategorysLoading: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
