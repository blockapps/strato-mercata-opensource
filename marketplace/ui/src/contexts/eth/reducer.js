import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.fetchETHSTAddress:
      return {
        ...state,
      };
    case actionDescriptors.fetchETHSTAddressSuccessful:
      return {
        ...state,
        ethstAddress: action.payload,
      };
    case actionDescriptors.fetchETHSTAddressFailed:
      return {
        ...state,
      };
    case actionDescriptors.addHash:
      return {
        ...state,
        isAddingHash: true,
      };
    case actionDescriptors.addHashSuccessful:
      return {
        ...state,
        isAddingHash: false,
      };
    case actionDescriptors.addHashFailed:
      return {
        ...state,
        isAddingHash: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
