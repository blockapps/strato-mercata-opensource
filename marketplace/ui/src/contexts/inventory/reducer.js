import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createInventory:
      return {
        ...state,
        isCreateInventorySubmitting: true,
      };
    case actionDescriptors.createInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.createInventoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.fetchInventory:
      return {
        ...state,
        isInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventorySuccessful:
      return {
        ...state,
        inventories: action.payload.data,
        inventoriesTotal: action.payload.count,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryFailed:
      return {
        ...state,
        error: action.error,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryForUser:
      return {
        ...state,
        isUserInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventoryForUserSuccessful:
      return {
        ...state,
        userInventories: action.payload.data,
        userInventoriesTotal: action.payload.count,
        isUserInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryForUserFailed:
      return {
        ...state,
        error: action.error,
        isUserInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventorySearch:
      return {
        ...state,
        isInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventorySearchSuccessful:
      return {
        ...state,
        inventories: action.payload.data,
        inventoriesTotal: action.payload.count,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventorySearchFailed:
      return {
        ...state,
        error: action.error,
        isInventoriesLoading: false,
      };
    case actionDescriptors.updateInventory:
      return {
        ...state,
        isinventoryUpdating: true,
      };
    case actionDescriptors.updateInventorySuccessful:
      return {
        ...state,
        inventoryUpdateObject: action.payload,
        isinventoryUpdating: false,
      };
    case actionDescriptors.updateInventoryFailed:
      return {
        ...state,
        error: action.error,
        isinventoryUpdating: false,
      };
    case actionDescriptors.updateSale:
      return {
        ...state,
        issaleUpdating: true,
      };
    case actionDescriptors.updateSaleSuccessful:
      return {
        ...state,
        saleUpdateObject: action.payload,
        issaleUpdating: false,
      };
    case actionDescriptors.updateSaleFailed:
      return {
        ...state,
        error: action.error,
        issaleUpdating: false,
      };
    case actionDescriptors.listInventory:
      return {
        ...state,
        isListing: true,
      };
    case actionDescriptors.listInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isListing: false,
      };
    case actionDescriptors.listInventoryFailed:
      return {
        ...state,
        error: action.error,
        isListing: false,
      };
    case actionDescriptors.unlistInventory:
      return {
        ...state,
        isUnlisting: true,
      };
    case actionDescriptors.unlistInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isUnlisting: false,
      };
    case actionDescriptors.unlistInventoryFailed:
      return {
        ...state,
        error: action.error,
        isUnlisting: false,
      };

     // ------------------------------------------------------------------------------------------------------
    case actionDescriptors.stakeInventory:
      return {
        ...state,
        isStaking: true
      };
    case actionDescriptors.stakeInventorySuccessful:
      return {
        ...state,
        isStaking: false
      };
    case actionDescriptors.stakeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isStaking: false
      };
    
    case actionDescriptors.unstakeInventory:
      return {
        ...state,
        isUnstaking: true
      };
    case actionDescriptors.unstakeInventorySuccessful:
      return {
        ...state,
        isUnstaking: false
      };
    case actionDescriptors.unstakeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isUnstaking: false
      };
    
    case actionDescriptors.borrow:
      return {
        ...state,
        isBorrowing: true
      };
    case actionDescriptors.borrowSuccessful:
      return {
        ...state,
        isBorrowing: false
      };
    case actionDescriptors.borrowFailed:
      return {
        ...state,
        error: action.error,
        isBorrowing: false
      };
    
    case actionDescriptors.repay:
      return {
        ...state,
        isRepaying: true
      };
    case actionDescriptors.repaySuccessful:
      return {
        ...state,
        isRepaying: false
      };
    case actionDescriptors.repayFailed:
        return {
          ...state,
          error: action.error,
          isRepaying: false
        };

    case actionDescriptors.getReserve:
        return {
          ...state,
          isReserveLoading: true
        };
    case actionDescriptors.getReserveSuccessful:
        return {
          ...state,
          reserve: action.payload,
          isReserveLoading: false
        };
    case actionDescriptors.getReserveFailed:
        return {
          ...state,
          error: action.error,
          isReserveLoading: false
        };
    
    case actionDescriptors.getAllReserve:
        return {
          ...state,
          isReservesLoading: true
        };
    case actionDescriptors.getAllReserveSuccessful:
        return {
          ...state,
          reserves: action.payload,
          isReservesLoading: false
        };
    case actionDescriptors.getAllReserveFailed:
        return {
          ...state,
          error: action.error,
          isReservesLoading: false
        };
    
    case actionDescriptors.getEscrowForAsset:
        return {
          ...state,
          isEscrowLoading: true
        };
    case actionDescriptors.getEscrowForAssetSuccessful:
        return {
          ...state,
          escrow: action.payload,
          isEscrowLoading: false
        };
    case actionDescriptors.getEscrowForAssetFailed:
        return {
          ...state,
          error: action.error,
          isEscrowLoading: false
        };

    case actionDescriptors.getUserCataRewards:
        return {
          ...state,
          isUserCataRewardsLoading: true
        };
    case actionDescriptors.getUserCataRewardsSuccessful:
        return {
          ...state,
          totalCataReward: action.payload.totalCataReward,
          dailyCataReward: action.payload.dailyCataReward,
          isUserCataRewardsLoading: false
        };
    case actionDescriptors.getUserCataRewardsFailed:
        return {
          ...state,
          error: action.error,
          isUserCataRewardsLoading: false
        };
    
    case actionDescriptors.getOracle:
          return {
            ...state,
            isFetchingOracle: true
          };
    case actionDescriptors.getOracleSuccessful:
          return {
            ...state,
            oracle: action.payload,
            isFetchingOracle: false
          };
    case actionDescriptors.getOracleFailed:
          return {
            ...state,
            error: action.error,
            isFetchingOracle: false
          };    
    // ------------------------------------------------------------------------------------------------------
    case actionDescriptors.resellInventory:
      return {
        ...state,
        isReselling: true,
      };
    case actionDescriptors.resellInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isReselling: false,
      };
    case actionDescriptors.resellInventoryFailed:
      return {
        ...state,
        error: action.error,
        isReselling: false,
      };
    case actionDescriptors.fetchSupportedTokens:
      return {
        ...state,
        isFetchingTokens: true,
      };
    case actionDescriptors.fetchSupportedTokensSuccessful:
      return {
        ...state,
        supportedTokens: action.payload,
        isFetchingTokens: false,
      };
    case actionDescriptors.fetchSupportedTokensFailed:
      return {
        ...state,
        error: action.error,
        isFetchingTokens: false,
      };
    case actionDescriptors.bridgeInventory:
      return {
        ...state,
        isBridging: true,
      };
    case actionDescriptors.bridgeInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isBridging: false,
      };
    case actionDescriptors.bridgeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isBridging: false,
      };
    case actionDescriptors.transferInventory:
      return {
        ...state,
        isTransferring: true,
      };
    case actionDescriptors.transferInventorySuccessful:
      return {
        ...state,
        isTransferring: false,
      };
    case actionDescriptors.transferInventoryFailed:
      return {
        ...state,
        error: action.error,
        isTransferring: false,
      };
    case actionDescriptors.fetchItemTransfers:
      return {
        ...state,
        isFetchingItemTransfers: true,
      };
    case actionDescriptors.fetchItemTransfersSuccessful:
      return {
        ...state,
        itemTransfers: action.payload.transfers,
        totalItemsTransfered: action.payload.total,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemTransfersFailed:
      return {
        ...state,
        error: action.error,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchInventoryOwnershipHistory:
      return {
        ...state,
        isInventoryOwnershipHistoryLoading: true,
      };
    case actionDescriptors.fetchInventoryOwnershipHistorySuccessful:
      return {
        ...state,
        inventoryOwnershipHistory: action.payload,
        isInventoryOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchInventoryOwnershipHistoryFailed:
      return {
        ...state,
        error: action.error,
        isInventoryOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchInventoryDetail:
      return {
        ...state,
        isInventoryDetailsLoading: true,
      };
    case actionDescriptors.fetchInventoryDetailSuccessful:
      return {
        ...state,
        inventoryDetails: action.payload,
        isInventoryDetailsLoading: false,
      };
    case actionDescriptors.fetchInventoryDetailFailed:
      return {
        ...state,
        error: action.error,
        isInventoryDetailsLoading: false,
      };
    case actionDescriptors.uploadImage:
      return {
        ...state,
        isUploadImageSubmitting: true,
      };
    case actionDescriptors.uploadImageSuccessful:
      return {
        ...state,
        uploadedImg: action.payload,
        isUploadImageSubmitting: false,
      };
    case actionDescriptors.uploadImageFailed:
      return {
        ...state,
        error: action.error,
        isUploadImageSubmitting: false,
      };
    case actionDescriptors.createItem:
      return {
        ...state,
        isCreateInventorySubmitting: true,
      };
    case actionDescriptors.createItemSuccessful:
      return {
        ...state,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.createItemFailed:
      return {
        ...state,
        error: action.error,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.fetchPriceHistory:
      return {
        ...state,
        isFetchingPriceHistory: true,
      };
    case actionDescriptors.fetchPriceHistorySuccessful:
      return {
        ...state,
        isFetchingPriceHistory: false,
        priceHistory: action.payload,
      };
    case actionDescriptors.fetchPriceHistoryFailed:
      return {
        ...state,
        isFetchingPriceHistory: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
