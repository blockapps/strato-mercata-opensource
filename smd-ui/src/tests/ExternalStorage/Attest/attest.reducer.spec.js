import reducer from "../../../components/ExternalStorage/Attest/attest.reducer";
import { initialState } from "../storageMock";
import { openAttestModal, closeAttestModal, attestDocument, attestDocumentSuccess, attestDocumentFailure, resetError, changeUsername } from "../../../components/ExternalStorage/Attest/attest.actions";
import { mockAttestFormData, error } from "./mockAttest";

describe('UploadFile: reducer', () => {

  // INITIAL_STATE
  test('set initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  describe('attest modal', () => {

    // OPEN_ATTEST_MODAL
    test('open', () => {
      const action = openAttestModal();
      expect(reducer(initialState, action)).toMatchSnapshot();
    })

    // CLOSE_ATTEST_MODAL
    test('close', () => {
      const action = closeAttestModal();
      expect(reducer(initialState, action)).toMatchSnapshot();
    })

  });

  describe('attest file', () => {

    // ATTEST_DOCUMENT_REQUEST
    test('request', () => {
      const action = attestDocument(mockAttestFormData);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // ATTEST_DOCUMENT_SUCCESS
    test('success', () => {
      const result = {
        "uri": "https://strato-external-storage.s3.amazonaws.com/1530165910145-widescreen.jpeg",
        "timeStamp": 1530165910,
        "signers": [
          "6e873015e8ff27d7c6d3ab5d1403a9df9ab420ad"
        ]
      };

      const action = attestDocumentSuccess(result);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // ATTEST_DOCUMENT_FAILURE
    test('failure', () => {
      const action = attestDocumentFailure(error);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

  });

  // RESET_ERROR
  test('reset error', () => {
    const init = {
      ...initialState,
      error: error
    }
    const action = resetError();
    expect(reducer(initialState, action)).toMatchSnapshot();
  });

  // USERNAME_FORM_CHANGE
  test('change username', () => {
    const init = {
      ...initialState,
      username: 'tanuj55'
    }

    const action = changeUsername();
    expect(reducer(initialState, action)).toMatchSnapshot();
  });


});