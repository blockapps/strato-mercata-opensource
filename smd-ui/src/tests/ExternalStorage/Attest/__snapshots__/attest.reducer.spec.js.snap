// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadFile: reducer attest file failure 1`] = `
Object {
  "attestDocument": null,
  "error": "ERROR",
  "isLoading": false,
  "uploadList": Array [],
}
`;

exports[`UploadFile: reducer attest file request 1`] = `
Object {
  "error": null,
  "isLoading": true,
  "uploadList": Array [],
}
`;

exports[`UploadFile: reducer attest file success 1`] = `
Object {
  "attestDocument": Object {
    "signers": Array [
      "6e873015e8ff27d7c6d3ab5d1403a9df9ab420ad",
    ],
    "timeStamp": 1530165910,
    "uri": "https://strato-external-storage.s3.amazonaws.com/1530165910145-widescreen.jpeg",
  },
  "error": null,
  "isLoading": false,
  "uploadList": Array [],
}
`;

exports[`UploadFile: reducer attest modal close 1`] = `
Object {
  "attestDocument": null,
  "error": null,
  "isOpen": false,
  "uploadList": Array [],
  "username": null,
}
`;

exports[`UploadFile: reducer attest modal open 1`] = `
Object {
  "error": null,
  "isOpen": true,
  "uploadList": Array [],
}
`;

exports[`UploadFile: reducer change username 1`] = `
Object {
  "error": null,
  "uploadList": Array [],
  "username": undefined,
}
`;

exports[`UploadFile: reducer reset error 1`] = `
Object {
  "error": null,
  "uploadList": Array [],
}
`;

exports[`UploadFile: reducer set initial state 1`] = `
Object {
  "attestDocument": null,
  "error": null,
  "isLoading": false,
  "isOpen": false,
  "username": null,
}
`;
